pkg/cmd/label/clone.go

```diff
@@ -105,32 +107,73 @@ func cloneRun(opts *cloneOptions) error {
 	return nil
 }
 
-func cloneLabels(client *http.Client, source, destination ghrepo.Interface, force bool) (successCount, totalCount int, err error) {
+func cloneLabels(client *http.Client, destination ghrepo.Interface, opts *cloneOptions) (successCount, totalCount int, err error) {
 	successCount = 0
-	labels, totalCount, err := listLabels(client, source, listQueryOptions{Limit: -1})
+	labels, totalCount, err := listLabels(client, opts.SourceRepo, listQueryOptions{Limit: -1})
 	if err != nil {
 		return
 	}
 
-	createOpts := createOptions{
-		Force: force,
+	workers := 10
+	toCreate := make(chan createOptions)
+	created := make(chan uint8)
```

Comment:
Nit: since the value doesn't matter, an idiomatic approach in Go would be to use an empty struct as value, just like you do with `doneCh`

Reply:
Idiomatic approach vs. non-idiomatic is always worth it. But you're right that atomic increment might be even simpler. Or, have workers do the increment themselves, where the increment is guarded by a mutex.

Reply:
What is the deadlock case in `gh status`?

Reply:
Exactly - was just thinking the workers would use `atomic.AddUint32`.

This is the same pattern that @samcoe used for `gh status` which works, except in some error cases the `errgroup.Group` (also: better/easier to use `errgroup.WithContext` as I did here since it handles the cancellation for you) terminates but the goroutine that is waiting to collect information from the workers continues looping and never exits. In tests this would time out. In various CIs for this PR, it repro'd a couple times, and locally it repro'd quite a few times. I think the anonymous goroutine polling for mentions in `gh status` needs to also check `<-ctx.Done()` to exit when the `errgroup` exits.

If no error happens it should all work fine. I was going to add a specific error case for `gh status` and see if I can repro the deadlock and fix it separately if you think this pattern - as it is now in the PR (I'll use `atomic` in this 1 case) - is fine.